### ISSUE TITLE
slic3r 1.2.0, 1.2.1, 1.2.2 respect min print speed / min layer time when in Spiral Vase Mode

### ISSUE THREAD

**Issue by SystemsGuy** [alexrj/Slic3r#2371]
Win64, but I don't think that matters.  When using spiral vase mode, it appears that slic3r is using the cooling parameters of "min_print_speed" and "slowdown_below_layer_time", resulting in a print speed that's equal to "min_print_speed" for the layer > bottom_solid_layers..

**Comment by bubnikv** [comment]
The original issue is bound to the calculation of layer time for spiral mode: The layer time is calculated for a single Z distance, which in spiral mode means only a short segment.

I believe the fix of the original issue has to do with the order of filters in the GCode.pm:
The _spiral_vase->process_layer is called before _cooling_buffer->append, meaning the Z coordinates are already spiralized. It seems to me that swapping the two filters would fix the problem.

**Comment by lordofhyphens** [comment]
@SystemsGuy @JBFromOZ @nophead 

I've made the change and applied it to a branch. Since there isn't a reproducible test case attached and I never use spiral vase mode, 

If you don't want to wait for the build to complete, grab a current build and open lib/Slic3r/Print/GCode.pm. 
Move lines 652-658 above 645. No rebuild is necessary, so you can just save the file and run Slic3r to see the difference (this may not work on the OSX build, as that is signed and I don't know if all the perl files are signed).

**Comment by alexrj** [comment]
@bubnikv, no, that's not true: short Z increments generated by SpiralVase are not considered as "layers" (that would be wrong). A new layer is supplied to CoolingBuffer whenever the extruder will extrude again on the same spot. In case of a spiral vase, each loop is a layer. So the logic is correct, and the original issue reported by @SystemsGuy is not a bug (especially since there's no test case proving a bug).

What emerged from the discussion is that slowdown is less needed for hollow vases, so there should be a way to turn it off without creating a new Filament profile, or implement some more sophisticated cooling logic which would take geometry into account.
